R/output_analysis.R

```diff
@@ -14,3 +14,29 @@ growth_rate <- function(out) {
   lambdas <- pops / c(NA, pops[1:(length(pops) - 1)])
   exp(mean(log(lambdas), na.rm = TRUE))
 }
+
+
+#' Calculate annual growth rate
+#'
+#' @param out Model output data frame
+#' @returns Numeric vector of length one representing the annual factor by
+#' which the total tick population changes. To use this function, it is best
+#' to run the model for at least three years.
+#'
+#' @examples
+#' out <- run(config_ex_1)
+#' annual_growth_rate(out)
+#'
+#' @export
+annual_growth_rate <- function(out) {
```

Comment:
It wouldn't hurt to first check that we have the columns we need. We could do something like this (which I tested locally):
```suggestion
annual_growth_rate <- function(out) {
if (!all(required_cols %in% names(out))) {
  missing_cols <- setdiff(required_cols, names(out))
  stop("Missing required colunns: ", paste(missing_cols, collapse = ", "))
}
```